vfs/dir.go

```diff
@@ -176,8 +197,8 @@ func (d *Dir) Node() Node {
 // so could not be forgotten. Children which didn't have virtual entries and
 // children with virtual entries will be forgotten even if true is returned.
 func (d *Dir) ForgetAll() (hasVirtual bool) {
-	d.mu.Lock()
-	defer d.mu.Unlock()
+	d.mu.RLock()
```

Comment:
Does this seem fine? Taking a Write lock was causing lockups for multiple minutes at once.

Reply:
Splitting the lock like that potentially makes the `hasVirtual` flag out of date.

Splitting this lock is a good idea though, I often see it in deadlock traces!

Maybe it would be better to keep the hasVirtual up to date continuously when we add and remove virtual entries so we don't have to deal with it here.

So when adding the first virtual entry you'd call hasVirtual(true) on the parent and it in turn would call its parents etc. Maybe the virtual flag could be read by sync/atomic to avoid the locking.